File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashSet;


public class Main {
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int n = Integer.parseInt(br.readLine());
		int count = 0;
		HashSet<String> hashSet = new HashSet<>();
		
		while(n--> 0) {
			String str = br.readLine();
			
			if(str.equals("ENTER")) {
				count += hashSet.size();
				hashSet.clear();
				continue;
			}
			
			hashSet.add(str);
		}
		
		System.out.println(count + hashSet.size());
	}
	
}

File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.cc
#include <iostream>
#include <set>

using namespace std;

int main()
{	
	int n, count = 0;

	cin >> n;

	set<string> s;
	string str;

	while (n-- > 0) {
		cin >> str;

		if (str == "ENTER") {
			count += s.size();
			s.clear();
			continue;
		}
		s.insert(str);
	}

	cout << count + s.size();

	return 0;
} 